### PR TITLE
V1.4 Assessment Feature

### PR DESCRIPTION
**Pull request by Meg1456**

### CODE REVIEW

**Comment by micklwl** [comment]
@Meg1456 Could you print the statistics list after each successful add/delete to the user as well? To ensure the lastShownStatsList stay updated as well.

**Comment by Meg1456** [comment]
> @Meg1456 Could you print the statistics list after each successful add/delete to the user as well? To ensure the lastShownStatsList stay updated as well.

Ok done!

File: docs/UserGuide.adoc [hunk 1]
```diff
@@ -724,4 +786,4 @@ a data saving operation. ClassRepo shuts itself down to ensure synchronisation o
 *Solution*: Ensure that the data files are not set as read-only. 
 
 == Known Bugs
-1. ClassRepo ASCII art may be formatted wrongly on OS's such as Mac.
+1. ClassRepo ASCII art may be incorrectly formatted on OS's such as Mac.
```
Comment (0WN463) [inline docs/UserGuide.adoc]: If possible, add in the image of the screen capture here

File: src/seedu/addressbook/commands/assessment/AddAssignmentStatistics.java [hunk 2]
```diff
@@ -44,10 +59,45 @@ public AssignmentStatistics getAssignmentStatistics() {
...
+            double total = 0;
+            int count = 0;
+            grade = assessName.getAllGrades();
+            for (Grades gradeVal : grade.values()) {
+                if (gradeVal.getValue() > maxGrade) {
+                    maxGrade = gradeVal.getValue();
+                }
+                if (gradeVal.getValue() < minGrade) {
+                    minGrade = gradeVal.getValue();
+                }
```
Comment (0WN463) [inline src/seedu/addressbook/commands/assessment/AddAssignmentStatistics.java]: Pretty sure Java have the max/min functions like in C++, could consider using them

File: src/seedu/addressbook/commands/assessment/AddAssignmentStatistics.java [hunk 3]
```diff
@@ -44,10 +59,45 @@ public AssignmentStatistics getAssignmentStatistics() {
...
+            }
+            if (count > 0) {
+                averageScore = (double) Math.round((total / count) * 100) / 100;
+            } else {
+                averageScore = 0; //to account for empty grades
+            }
+            totalExamTakers = count;
+            maxScore = maxGrade;
+            if (minGrade == 1000000) { //to account for empty grades
+                minScore = 0;
```
Comment (0WN463) [inline src/seedu/addressbook/commands/assessment/AddAssignmentStatistics.java]: Seems a bit potato

File: src/seedu/addressbook/common/Messages.java [hunk 4]
```diff
@@ -23,6 +27,8 @@
     public static final String MESSAGE_USING_EXAMS_FILE = "Using exams file : %1$s";
     public static final String MESSAGE_EXAMS_LISTED_OVERVIEW = "%1$d exams listed!";
     public static final String MESSAGE_USING_STATISTICS_FILE = "Using statistics file : %1$s";
+    public static final String MESSAGE_STATISTICS_LISTED_OVERVIEW = "%1$d statistics listed!";
+    public static final String MESSAGE_INVALID_GRADES = "Invalid command format! - Value of grades is out of bounds!";
     public static final String MESSAGE_EXAM_NOT_IN_EXAMBOOK = "Exam could not be found in exam book!";
```
Comment (0WN463) [inline src/seedu/addressbook/common/Messages.java]: Seems a bit bad naming. Out of bounds implies it is an index, but it says command format invalid

File: test/java/seedu/addressbook/TestDataHelper.java [hunk 5]
```diff
@@ -59,19 +61,14 @@ public Fees fees (int seed) throws Exception {
         return new Fees(value, dueDate);
     }
 
-    /** Test exam for testing**/
+    /** Test statistics for testing**/
     public AssignmentStatistics stat() throws Exception {
-        String subjectName = "Spanish";
```
Comment (0WN463) [inline test/java/seedu/addressbook/TestDataHelper.java]: Methods should sound like verbs

File: test/java/seedu/addressbook/parser/ParserTest.java [hunk 6]
```diff
@@ -197,6 +223,23 @@ public void deleteGradesCommand_noArgs() {
...
+
+    //    @Test
+    //    public void deleteGradesCommand_numericArg_indexParsedCorrectly() {
+    //        final int testIndex1 = 1;
+    //        final int testIndex2 = 2;
+    //        final String input = "deletegrades " + testIndex1 + testIndex2;
+    //        final DeleteGradesCommand result = parseAndAssertCommandType(input, DeleteGradesCommand.class);
+    //        assertEquals(result.getTargetIndex(ObjectTargeted.PERSON), testIndex1);
+    //    }
+
```
Comment (0WN463) [inline test/java/seedu/addressbook/parser/ParserTest.java]: Remove if unused

File: test/java/seedu/addressbook/TestDataHelper.java (hunk 5 above)
Comment (micklwl) [reply test/java/seedu/addressbook/TestDataHelper.java]: Help to add "make" for the other object creators as well. eg. makeStat, makeAdam

**Pull request merged**